src/Microsoft.AspNet.Mvc.Extensions/Formatters/JsonOutputFormatter.cs

```diff
@@ -55,7 +65,12 @@ public void WriteObject([NotNull] TextWriter writer, object value)
             }
         }
 
-        private JsonWriter CreateJsonWriter(TextWriter writer)
+        /// <summary>
+        /// Called during serialization to get the <see cref="JsonWriter"/>.
+        /// </summary>
+        /// <param name="writer">The <see cref="TextWriter"/> used to write.</param>
+        /// <returns>The <see cref="JsonWriter"/> used during serialization.</returns>
+        public virtual JsonWriter CreateJsonWriter(TextWriter writer)
```

Comment:
Not sure why this is public and not protected. What's the deal with `JsonInputFormatter`? Can you track down the reason/history why it's public there?